lib/taste_tester/commands.rb

```diff
@@ -106,6 +106,21 @@ def self.test
         TasteTester::Hooks.post_test(TasteTester::Config.dryrun, repo,
                                      tested_hosts)
       end
+      # strictly, hosts and tested_hosts should be sets to eliminate
+      # variance in order or duplicates. The exact comparison works
+      # because we're building the list in the same order
+      if tested_hosts == hosts
+        # no exceptions, complete success
+        exit(0)
+      end
+      if tested_hosts.empty?
+        # nothing worked, complete failure
+        exit(3)
+      end
+      # otherwise, we got a mix of success and failure. This isn't well
```

Comment:
This is highly misleading.

These are ones we *chose* not to test because they were already in testing.

That's very different from failed to test. If we fail to test any host, we immediately crash/bail out (if the host is down, if you don't have ssh access, etc.)

I'm OK with a different exit code here... but both the comments here and the documentation you added above is very misleading.

Reply:
If you taste-test one or more hosts, and the end result is that the taste-testing state of the selected hosts does not change then the test command did not succeed.

host.in_test? is specifically asserting the user is the same. If someone else is testing, then you are not testing.

Other failure modes like inability to connect are also fatal, but there are no changes to this.